clientd3d/music.c

```diff
@@ -206,29 +258,27 @@ DWORD PlayMidiFile(HWND hWndNotify, char *fname)
 		debug(( "Failed to init music file.\n" ));
 		return 0;
 	}
-
+   // Loop the MIDI until game actions stop it
+	AIL_set_sample_loop_count( hseqImmediate, 0 );
    // Set volume
    float vol = ((float) config.music_volume) / CONFIG_MAX_VOLUME;
 	AIL_set_sample_volume_levels(hseqImmediate, vol, vol );
 
 	// start playing
 	AIL_start_sample(hseqImmediate);
 
-	// Set end-of-sample callback so we can unpause
-	//	the background music when done playing.
-	AIL_register_EOS_callback(hseqImmediate, MIDIDoneCallback);
-
-	debug(( "Playing music file %s.\n", fname ));
+	debug(( "Playing midi music file %s. LatestMusic=%d \n", fname, latest_music ));
 	playing_midi = True;
 	return 0;
 #else
    {
       DWORD dwReturn;
       MCI_PLAY_PARMS mciPlayParms;
-
-      if ((dwReturn = OpenMidiFile(fname)) != 0)
+      // Open file with deviceflag = 2
+      if ((dwReturn = OpenMidiFile(fname, 2)) != 0)
       {
-        return dwReturn;
+         debug(("OpenMidiFile error - can't open \n"));
```

Comment:
This debug message wouldn't help very much--at least include the filename.

Reply:
Thanks, done.